website/addons/osfstorage/views.py

```diff
@@ -229,3 +229,28 @@ def osfstorage_download(file_node, payload, node_addon, **kwargs):
             osf_storage_settings.WATERBUTLER_RESOURCE: version.location[osf_storage_settings.WATERBUTLER_RESOURCE],
         },
     }
+
+@must_have_permission('write')
+@decorators.autoload_filenode(must_be='file')
+def osfstorage_add_tag(file_node, **kwargs):
```

Comment:
I have done a functional test and can confirm that files in a registration cannot have tags attached to them